documentation/API-GUIDELINES.md

```diff
@@ -86,6 +86,7 @@ In general, the [Rust API Guidelines](https://rust-lang.github.io/api-guidelines
 - Every line of code is a liability. Take some time to see if your implementation can be simplified before opening a PR.
 - If you are porting code from ESP-IDF (or anything else), please include a link WITH the commit hash in it, and please highlight the relevant line(s) of code
 - If necessary provide further context as comments (consider linking to code, PRs, TRM - make sure to use permanent links, e.g. include the hash when linking to a Git repository, include the revision, page number etc. when linking to TRMs)
+- Prefer line comments (//) to block comments (/* ... */)
```

Comment:
Should I add a link to the [style guide](https://doc.rust-lang.org/nightly/style-guide/#comments) here?

Reply:
It's a perfect opportunity to rename the file from API-GUIDELINEs, as there are a bunch of things in there that are independent of the actual API.

Reply:
Would something like "DEVELOPMENT GUIDELINEs" sound better?

Reply:
maybe just `CODE-GUIDELINES.md`?

at some point we also might want to split the common things applying to all crates and things related to esp-hal drivers (maybe)

Reply:
I guess we can deal with deformatting the doc later :)